esi_leap/resource_objects/ironic_node.py

```diff
@@ -34,34 +34,33 @@ class IronicNode(base.ResourceObjectInterface):
 
     def __init__(self, ident):
         if not is_uuid_like(ident):
-            node = get_ironic_client().node.get(ident)
-            ident = node.uuid
-        self._uuid = ident
+            self._node = get_ironic_client().node.get(ident)
+            self._uuid = self._node.uuid
+        else:
+            self._node = None
+            self._uuid = ident
 
     def get_resource_uuid(self):
         return self._uuid
 
     def get_resource_name(self, resource_list=None):
-        node = ironic.get_node(self._uuid, resource_list)
-        return getattr(node, 'name', '')
+        return getattr(self._get_node(resource_list), 'name', '')
 
     def get_lease_uuid(self):
-        node = get_ironic_client().node.get(self._uuid)
-        return node.properties.get('lease_uuid', None)
+        return self._get_node().properties.get('lease_uuid', None)
 
     def get_project_id(self):
-        node = get_ironic_client().node.get(self._uuid)
-        return node.lessee
+        return self._get_node().lessee
 
     def get_node_config(self):
```

Comment:
Not relevant to this PR, but I don't see this function is called anywhere.

Reply:
Weird, you're right - I'm not sure what it was used for either!